tough_control/tough_kinematics/src/tough_kinematics.cpp

```diff
@@ -0,0 +1,206 @@
+#include <tough_kinematics/tough_kinematics.h>
+
+ToughKinematics::ToughKinematics(ros::NodeHandle& nh) : nh_(nh)
+{
+  state_informer_ = RobotStateInformer::getRobotStateInformer(nh);
+  rd_ = RobotDescription::getRobotDescription(nh);
+
+  addExistingKDLChains();
+}
+
+ToughKinematics::~ToughKinematics()
+{
+  for(auto i:ik_solvers_)
```

Comment:
use `auto &`. more details [here](https://stackoverflow.com/questions/15927033/what-is-the-correct-way-of-using-c11s-range-based-for)